pkg/iris/mapping.go

```diff
@@ -34,45 +41,102 @@ func NewMapper(
 	}
 }
 
-func (m *mapper) GetDeployments() ([]DiscoveryItem, error) {
-	namespaces, err := m.KubernetesApi.Namespaces(m.BlackListedNamespaces)
-	if err != nil {
-		return nil, err
+type ClusterDTO struct {
+	name       string
+	k8sVersion string
+	nodesCount int
+	osImage    string
+}
+
+// GetCluster MapNodes maps a list of nodes and a given cluster name into a KubernetesObject.
+// In the process it aggregates the information from muliple nodes into one cluster object.
+func (m *mapper) GetCluster(clusterName string, nodes *v1.NodeList) (ClusterDTO, error) {
+
+	items := nodes.Items
+	if len(items) == 0 {
+		return ClusterDTO{
+			name: clusterName,
+		}, nil
 	}
-	var allDiscoveryItems []DiscoveryItem
-	for _, namespace := range namespaces.Items {
-		// complexity is O(mxn)
-		if blacklistedNS(namespace.Name, m.BlackListedNamespaces) {
-			continue
-		}
-		log.Infof("Fetching Deployments for namespace [%s]", namespace.Name)
-		deployments, err := m.KubernetesApi.Deployments(namespace.Name)
-		if err != nil {
-			return nil, err
-		}
-		log.Infof("mapping deployments for namespace [%s]", namespace.Name)
-		mappedDeployments := m.MapDeployments(deployments)
-		allDiscoveryItems = append(allDiscoveryItems, mappedDeployments...)
+	k8sVersion := set.NewStringSet()
+	osImage := set.NewStringSet()
+
+	for _, n := range items {
+		k8sVersion.Add(n.Status.NodeInfo.KubeletVersion)
+		osImage.Add(n.Status.NodeInfo.OSImage)
+
 	}
-	log.Info("Fetching deployments for namespaces completed")
-	return allDiscoveryItems, nil
+	return ClusterDTO{
+		name:       clusterName,
+		k8sVersion: strings.Join(k8sVersion.Items(), ", "),
+		nodesCount: len(items),
+		osImage:    strings.Join(osImage.Items(), ", "),
+	}, nil
 }
 
-func (m *mapper) MapDeployments(deployments *appsv1.DeploymentList) []DiscoveryItem {
-	var deploymentDiscoveryItems []DiscoveryItem
+func (m *mapper) GetDeployments(deployments *appsv1.DeploymentList, services *v1.ServiceList) ([]models.Deployment, error) {
```

Comment:
Just a thought. `MapDeployment` is the right name than `GetDeployment` ?

Reply:
absolutely 👍